Makefile

```diff
@@ -130,9 +131,24 @@ src/nvim/testdir/%.vim: phony_force
 functionaltest functionaltest-lua unittest benchmark: | nvim
 	$(BUILD_TOOL) -C build $@
 
-lintlua lintsh lintpy lintuncrustify lintc lintcfull check-single-includes generated-sources lintcommit lint formatc formatlua format: | build/.ran-cmake
+check-single-includes generated-sources formatc formatlua format: | build/.ran-cmake
 	$(CMAKE_PRG) --build build --target $@
 
+build/CTestTestfile.cmake:
+	$(CMAKE_PRG) -S . -B build
+
+lintlua lintsh lintpy lintcommit: build/CTestTestfile.cmake
+	ctest --test-dir build $(CTEST_EXTRA_FLAG) -R $@
+
+lint: build/CTestTestfile.cmake
+	ctest --test-dir build $(CTEST_EXTRA_FLAG) -L 'lint-all'
+
+lintc: build/CTestTestfile.cmake check-single-includes
+	ctest --test-dir build $(CTEST_EXTRA_FLAG) -R lintc-changed-only
+
+lintcfull: build/CTestTestfile.cmake
+	ctest --test-dir build $(CTEST_EXTRA_FLAG) -L 'lintcfull'
```

Comment:
Shouldn't these have `-j` to parallelize them?

Reply:
oop, I forgot to set [`$CTEST_PARALLEL_LEVEL`](https://cmake.org/cmake/help/latest/envvar/CTEST_PARALLEL_LEVEL.html)